pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -2,50 +2,111 @@ package elasticsearch
+}
+
+func (s *Service) Init() error {
+	eslog.Debug("initializing")
+	im := datasource.NewInstanceManager(newInstanceSettings())
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: newService(im, s.Cfg, s.HTTPClientProvider),
+	})
+	if err := s.BackendPluginManager.Register("elasticsearch", factory); err != nil {
+		eslog.Error("Failed to register plugin", "error", err)
+	}
+	return nil
 }
 
-// New creates a new Executor func.
-// nolint:staticcheck // plugins.DataPlugin deprecated
-func New(httpClientProvider httpclient.Provider) func(*models.DataSource) (plugins.DataPlugin, error) {
-	// nolint:staticcheck // plugins.DataPlugin deprecated
-	return func(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-		return &Executor{
-			httpClientProvider: httpClientProvider,
-			intervalCalculator: interval.NewCalculator(),
-		}, nil
+// newService creates a new executor func.
+func newService(im instancemgmt.InstanceManager, cfg *setting.Cfg, httpClientProvider httpclient.Provider) *Service {
+	return &Service{
+		im:                 im,
+		cfg:                cfg,
+		HTTPClientProvider: httpClientProvider,
+		intervalCalculator: tsdb.NewCalculator(),
 	}
 }
 
-// Query handles an elasticsearch datasource request
-//nolint: staticcheck // plugins.DataResponse deprecated
-func (e *Executor) DataQuery(ctx context.Context, dsInfo *models.DataSource,
-	tsdbQuery plugins.DataQuery) (plugins.DataResponse, error) {
-	if len(tsdbQuery.Queries) == 0 {
-		return plugins.DataResponse{}, fmt.Errorf("query contains no queries")
+func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	if len(req.Queries) == 0 {
+		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
 	}
 
-	client, err := es.NewClient(ctx, e.httpClientProvider, dsInfo, *tsdbQuery.TimeRange)
+	dsInfo, err := s.getDSInfo(req.PluginContext)
 	if err != nil {
-		return plugins.DataResponse{}, err
+		return &backend.QueryDataResponse{}, err
 	}
 
-	if tsdbQuery.Debug {
-		client.EnableDebug()
+	client, err := es.NewClient(ctx, s.HTTPClientProvider, dsInfo, req.Queries[0].TimeRange)
+	if err != nil {
+		return &backend.QueryDataResponse{}, err
 	}
 
-	query := newTimeSeriesQuery(client, tsdbQuery, e.intervalCalculator)
+	query := newTimeSeriesQuery(client, req.Queries, s.intervalCalculator)
 	return query.execute()
 }
+
+func newInstanceSettings() datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		jsonData := map[string]interface{}{}
+		err := json.Unmarshal(settings.JSONData, &jsonData)
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+		httpCliOpts, err := settings.HTTPClientOptions()
+		if err != nil {
+			return nil, fmt.Errorf("error getting http options: %w", err)
+		}
+		model := es.DatasourceInfo{
+			JsonData:    simplejson.NewFromAny(jsonData),
```

Comment:
Think you can use and skip the unmarshal to `map[string]interface{}`.

```suggestion
			JsonData:    simplejson.NewJson(settings.JSONData),
```